ttnn/cpp/ttnn/operations/data_movement/pad/device/pad_program_factory.cpp

```diff
@@ -1575,4 +1577,145 @@ operation::ProgramWithCallbacks pad_rm_sharded(
     return {.program = std::move(program), .override_runtime_arguments_callback = override_runtime_args_callback};
 }
 
+operation::ProgramWithCallbacks pad_rm_sharded_width_only(
+    const Tensor& input_tensor,
+    Tensor& output,
+    const tt::tt_metal::LegacyShape& output_tensor_shape,
+    const ttnn::SimpleShape& input_tensor_start,
+    float pad_value) {
+    Program program{};
+
+    TT_ASSERT(
+        output.shard_spec().has_value() and output.shard_spec()->shape[1] == output_tensor_shape[-1],
+        "output must be sharded with shard width equal to output width");
+
+    uint32_t W = input_tensor.shape()[3], H = input_tensor.shape()[2], C = input_tensor.shape()[1], N = input_tensor.shape()[0];
+    uint32_t W_padded = output_tensor_shape[3], H_padded = output_tensor_shape[2], C_padded = output_tensor_shape[1], N_padded = output_tensor_shape[0];
```

Comment:
N_padded is unused. Please check that all declared variables are actually used.

Reply:
(side note): we must enable compiler warning for unused varibles, but to do that we'd have to fix everything. 
@blozano-tt @afuller-TT I wonder if there is any clang-tidy check and autofixer for it

Reply:
I haven't seen any auto-fixer for that particular thing.  Would probably require some some manual intervention.  But it's easy -- delete! 😄

Reply:
Thanks for catching this. I don't have working clangd right now so I'm basically just looking for unused variables manually 😢